src/vs/base/parts/storage/common/storage.ts

```diff
@@ -278,7 +283,7 @@ export class Storage extends Disposable implements IStorage {
 		this.pendingInserts.delete(key);
 
 		// Event
-		this._onDidChangeStorage.fire(key);
+		this._onDidChangeStorage.fire({ key, wasChangedExternally: false });
```

Comment:
Do we need to allow to pass in `wasChangedExternally` also for `delete`? What if a settings sync removes a key because its value changed to a default value?